packages/core/src/components/axes/axis.ts

```diff
@@ -70,13 +70,44 @@ export class Axis extends Component {
 				break;
 		}
 
+		// Add axis into the parent
+		const container = DOMUtils.appendOrSelect(svg, `g.axis.${axisPosition}`);
+		const axisRefExists = !container.select(`g.ticks`).empty();
+		let axisRef = DOMUtils.appendOrSelect(container, `g.ticks`);
+		if (!axisRefExists) {
+			axisRef.attr("role", `${Roles.GRAPHICS_OBJECT} ${Roles.GROUP}`);
+		}
+
+		// We draw the invisible axis because of the async nature of d3 transitions
+		// To be able to tell the final width & height of the axis when initiaing the transition
+		// The invisible axis is updated instantly and without a transition
+		const invisibleAxisRef = DOMUtils.appendOrSelect(container, `g.ticks.invisible`)
+			.style("opacity", "0")
+			.attr("aria-hidden", true);
+
+		// Get font height
+		const fakeTick = DOMUtils.appendOrSelect(invisibleAxisRef, `g.tick`);
+		const fakeTickText = DOMUtils.appendOrSelect(fakeTick, `text`).text("0");
+		const tickHeight = fakeTickText.node().getBBox().height;
+		fakeTick.remove();
+
 		// Initialize axis object
 		const axis = axisFunction(scale)
 			.tickSizeOuter(0)
 			.tickFormat(Tools.getProperty(axisOptions, "ticks", "formatter"));
 
 		if (scale.ticks) {
-			const numberOfTicks = Tools.getProperty(axisOptions, "ticks", "number") || Configuration.axis.ticks.number;
+			let numberOfTicks = Tools.getProperty(axisOptions, "ticks", "number");
```

Comment:
You'd still need `|| Configuration.axis.ticks.number`. User-provided options normally are merged with our default configurations, except for `axes` options (because of the limitations that'd apply there) so we'd still need to fallback on a couple of things

Reply:
This was implemented in this way to follow this logic:
 - if the user does specify an `options.[axis].ticks.number`, that tickNumber is used (ignoring the chart height)
 - if the user does NOT specify the option, the `Configuration.axis.ticks.number` is used as a default but it's adapted on the chart height.

The fallback to `|| Configuration.axis.ticks.number` is done in line 108.

Are you suggesting a different logic? 
We could make it more obvious, maybe a boolean `adaptTicksNumberBasedOnHeight`, but we feel it would be less easy to use.

Reply:
@theiliad any feedback on this point? Just to understand if I misunderstood